lib/actuator/include/actuator/LiftsLimitSwitchesManager.hpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) CONFIG_ACTUATOR_NUMBER25 COGIP Robotics association <cogip35@gmail.com>
+// This file is subject to the terms and conditions of the GNU Lesser
+// General Public License v2.1. See the file LICENSE in the top level
+// directory for more details.
+
+/// @ingroup     actuator
+/// @{
+/// @file
+/// @brief       Singleton managing GPIO-based limit switches for motors.
+/// @author      Gilles DOFFE <g.doffe@gmail.com>
+
+#pragma once
+
+// RIOT includes
+#include <event.h>
+#include <periph/gpio.h>
+
+// Project includes
+#include "actuator/Lift.hpp"
+
+// Packages includes
+#include "etl/map.h"
+#include "etl/delegate.h"
+#include "etl/pool.h"
+
+
+namespace cogip {
+namespace actuators {
+namespace positional_actuators {
+
+/// @brief Singleton managing GPIO-based limit switches for motors.
+/// @details This class handles GPIO interrupts via a dedicated thread,
+///          allowing motor control to react to events such as limit switches
+///          in a safe, non-blocking, and asynchronous way.
+class LiftsLimitSwitchesManager {
+public:
+    /// @brief Callback type used to react to GPIO events in thread context.
+    using GpioCallback = etl::delegate<void(void)>;
+
+    /// @brief Get the singleton instance of the manager.
+    /// @return Reference to the LiftsLimitSwitchesManager singleton.
+    static LiftsLimitSwitchesManager& instance();
+
+    /// @brief Initialize the manager by creating the event thread and queue.
+    void init();
+
+    /// @brief Register a GPIO pin for falling-edge interrupts with a Lift handler.
+    /// @details Configures the pin with pulldown and sets up the ISR to enqueue
+    ///          events which are later dispatched in thread context.
+    /// @param pin GPIO pin to register.
+    /// @param lift Pointer to the Lift actuator instance to notify on trigger.
+    void register_gpio(gpio_t pin, cogip::actuators::positional_actuators::Lift* lift);
+
+    /// @brief Event handler executed in the event thread context.
+    /// @param evt Pointer to the event to handle.
+    static void event_handler(event_t* evt);
+
+private:
+    /// @brief Private constructor to enforce singleton pattern.
+    LiftsLimitSwitchesManager() = default;
+
+    /// @brief GPIO ISR callback invoked in interrupt context.
+    /// @details Enqueues the corresponding event for later handling.
+    /// @param arg Pointer to the GPIO pin wrapped as void*.
+    static void isr_callback(void* arg);
+
+    /// @brief Entry function for the event processing thread.
+    /// @param arg Unused thread argument.
+    /// @return nullptr on exit.
+    static void* event_thread_entry(void* arg);
+
+    /// @brief Dispatches a GPIO event to the registered Lift callback.
+    /// @param pin GPIO pin whose event occurred.
+    void handle_event(gpio_t pin);
```

Comment:
Maybe it can be renamed to `handle_gpio_event` to avoid confusing with `event_handler` as their names are close.

Reply:
Done.